prow/tide/history/history.go

```diff
@@ -45,11 +45,17 @@ type History struct {
 	sync.Mutex
 	logSizeLimit int
 
-	opener io.Opener
+	opener opener
 	path   string
 }
 
-func readHistory(maxRecordsPerKey int, opener io.Opener, path string) (map[string]*recordLog, error) {
+// opener has methods to read and write paths
+type opener interface {
```

Comment:
Could you move all the consumer definitions of the `opener` interface from this PR into a dedicated one that we can merge already to keep the size of this a bit more manageable?

Reply:
Done > https://github.com/kubernetes/test-infra/pull/17301